static/app/views/detectors/components/details/metric/chart.tsx

```diff
@@ -0,0 +1,51 @@
+import styled from '@emotion/styled';
+
+import type {MetricDetector} from 'sentry/types/workflowEngine/detectors';
+import {MetricDetectorChart} from 'sentry/views/detectors/components/forms/metric/metricDetectorChart';
+import {getDetectorDataset} from 'sentry/views/detectors/components/forms/metric/metricFormData';
+
+interface MetricDetectorDetailsChartProps {
+  detector: MetricDetector;
+}
+
+export function MetricDetectorDetailsChart({detector}: MetricDetectorDetailsChartProps) {
+  const dataSource = detector.dataSources[0];
+  const snubaQuery =
+    dataSource?.type === 'snuba_query_subscription'
+      ? dataSource.queryObj?.snubaQuery
+      : undefined;
+
+  if (!snubaQuery) {
+    // Unlikely, helps narrow types
```

Comment:
For the uptime types I made this a tuple so that we don't need to keep checking for this. Could also do that for metric issues, though I don't know how safe/unsafe it is to assume it's always there